### DISCUSSION TITLE
Rails 7 - Format y-axis to dollars

### DISCUSSION THREAD

**Discussion by bweathers3** [styd/apexcharts.rb#81] [Q&A] [answered]
I am trying to to format my y-axis key to be USA dollars rather than just a number value.

My code in the view is as follows:

```
<h2>Revenue vs Expenses</h2>
​
<% series = [
  {name: "Revenue", data: @rev_chart},
  {name: "Expenses", data: @exp_chart}
] %>
​
<% options = {
  title: 'Revenue vs Expenses',
  subtitle: 'Grouped Per Month',
  xtitle: 'Months',
  ytitle: 'Dollars',
} %>

<%= column_chart(series, {**options, theme: 'overview_bar_chart_palette', stacked: false, data_labels: false}) %>

```

The initializer code:

```
ApexCharts::Theme.create "overview_bar_chart_palette", ["#008FFB", "#FF9800"]

```

So basically  100.00, 200.00, 300.00 etc.  would be formatted to   $100.00, $200.00, $300.00 on the y-axis key.

**Comment by styd** [comment, marked as answer]
Hi, @bweathers3 . Sorry for a late reply. I was on a vacation.

Have you tried using [formatter function](https://github.com/styd/apexcharts.rb#formatter-function)?
I think you can try adding formatter for y-axis' label:
```erb
<% options = {
  title: 'Revenue vs Expenses',
  subtitle: 'Grouped Per Month',
  xtitle: 'Months',
  ytitle: 'Dollars',
  yaxis: {
    labels: {
      formatter: {
        function: {
          args: "val",
          body: "return '$' + parseFloat(val).toLocaleString();"
        }
      }
    }
  } 
} %>
```
You can also do so for the dataLabels.

**Comment by bweathers3** [comment]
Adrian,

Thanks so much, that worked perfectly.

Bailey
